executor.go

```diff
@@ -5,30 +5,49 @@ import (
 	"io"
 )
 
-type executor struct {
-	output      io.Writer
-	defined     map[string]*taskSnapshot
-	logger      Logger
-	middlewares []Middleware
-	noDeps      bool
-}
+type (
+	// Executor represents a flow execution function.
+	Executor func(ExecuteInput) error
+
+	// ExecuteInput received by the flow executor.
+	ExecuteInput struct {
+		Context   context.Context
+		Tasks     []string
+		SkipTasks []string
+		NoDeps    bool
+		Output    io.Writer
+		Logger    Logger
+	}
+
+	// ExecutorMiddleware represents a flow execution interceptor.
+	ExecutorMiddleware func(Executor) Executor
+
+	executor struct {
+		defined     map[string]*taskSnapshot
+		middlewares []Middleware
+	}
+)
 
 // Execute runs provided tasks and all their dependencies.
 // Each task is executed at most once.
```

Comment:
I would add this method is there to satisfy/implements Executor interface

Reply:
Not needed as other code would not compile.